Subject: sched: Fix up /proc/sched_debug to print only runnable tasks again
From: Mike Galbraith <bitbucket@online.de>
Date: Tue Jun  3 14:16:09 CEST 2014
Patch-mainline: never, SUSE specific (submitted, but ignored)

Commit b32e86b430 dropped p->on_rq check, leading to printing tasks which
used to live on runqueues, which is very noisy and highly annoying.

Signed-off-by: Mike Galbraith <mgalbraith@suse.de>
---
 kernel/sched/debug.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/kernel/sched/debug.c
+++ b/kernel/sched/debug.c
@@ -463,7 +463,7 @@ static void print_rq(struct seq_file *m,
 
 	rcu_read_lock();
 	for_each_process_thread(g, p) {
-		if (task_cpu(p) != rq_cpu)
+		if (!p->on_rq || task_cpu(p) != rq_cpu)
 			continue;
 
 		print_task(m, rq, p);
